src/uct/ib/rc/accel/rc_mlx5_common.c

```diff
@@ -174,13 +211,115 @@ unsigned uct_rc_mlx5_iface_srq_post_recv(uct_rc_mlx5_iface_common_t *iface)
     return count;
 }
 
+unsigned uct_rc_mlx5_iface_srq_post_recv_sge(uct_rc_mlx5_iface_common_t *iface)
+{
+    uct_ib_mlx5_srq_t *srq   = &iface->rx.srq;
+    uct_rc_iface_t *rc_iface = &iface->super;
+    uct_ib_mlx5_srq_seg_t *seg;
+    uint16_t count, wqe_index, next_index;
+    ucs_status_t status;
+    uct_user_allocator_buffs_t rx_buffers;
+    uint32_t buff_idx;
+
+    /* Make sure the union is right */
+    uct_rc_mlx5_iface_srq_post_recv_check_union(iface);
+
+    wqe_index = srq->ready_idx;
+    buff_idx  = 0;
+    //TODO - check with Yossi if we can replace loop with some direct calculation
+    for (;;) {
+        next_index = wqe_index + 1;
+        seg        = uct_ib_mlx5_srq_get_wqe(srq, next_index);
+        if (UCS_CIRCULAR_COMPARE16(next_index, >, srq->free_idx)) {
+            if (!seg->srq.free) {
+                break;
+            }
+        }
+        wqe_index = next_index;
+        buff_idx++;
+    }
+    rx_buffers.num_of_buffers = buff_idx;
+    UCT_TL_IFACE_GET_RX_DESC_SG(&iface->super.super.super, iface->super.rx.mps,
```

Comment:
maybe keep this macro as is and implement it as part of mlx5 post recv code only